### PR TITLE
Add Missing Type Annotations

### LINKED ISSUE #20: Missing Type Annotations
There are missing type annotations for some functions in this library.

The `typing` module does not exist on CircuitPython devices so the import needs to be wrapped in try/except to catch the error for missing import. There is an example of how that is done here:

```python
try:
    from typing import List, Tuple
except ImportError:
    pass
```

Once imported the typing annotations for the argument type(s), and return type(s) can be added to the function signature. Here is an example of a function that has had this done already:

```python
def wrap_text_to_pixels(
    string: str, max_width: int, font=None, indent0: str = "", indent1: str = ""
) -> List[str]:
```
If you are new to Git or Github we have a guide about contributing to our projects here: https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github

There is also a guide that covers our CI utilities and how to run them locally to ensure they will pass in Github Actions here: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/check-your-code In particular the pages: `Sharing docs on ReadTheDocs` and `Check your code with pre-commit` contain the tools to install and commands to run locally to run the checks.

If you are attempting to resolve this issue and need help, you can post a comment on this issue and tag both @foamyguy and @kattni or reach out to us on Discord: https://adafru.it/discord in the `#circuitpython-dev` channel.

The following locations are reported by mypy to be missing type annotations:

- [ ] adafruit_stmpe610.py:162
- [ ] adafruit_stmpe610.py:166
- [ ] adafruit_stmpe610.py:172
- [ ] adafruit_stmpe610.py:229
- [ ] adafruit_stmpe610.py:243
- [ ] adafruit_stmpe610.py:252
- [ ] adafruit_stmpe610.py:264
- [ ] adafruit_stmpe610.py:289
- [ ] adafruit_stmpe610.py:299

### PR DESCRIPTION
**Pull request by tcfranks**
resolves #20 
submitted for review / comment

### CODE REVIEW

File: adafruit_stmpe610.py [hunk 1]
```diff
@@ -26,12 +26,21 @@
...
+except ImportError:
+    pass
 
 __version__ = "0.0.0+auto.0"
 __repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_STMPE610.git"
 
 
-def map_range(x, in_min, in_max, out_min, out_max):
+def map_range(
+    x: Union[int, float], in_min: int, in_max: int, out_min: int, out_max: int
```
Comment (tekktrik) [inline adafruit_stmpe610.py]: When typing something as a `float` it is assumed `int` may also be used.  Essentially, `float` can be used in place of `Union[int, float]`.  See [PEP 484](https://peps.python.org/pep-0484/#the-numeric-tower) for more information.

File: adafruit_stmpe610.py [hunk 2]
```diff
@@ -199,23 +208,23 @@ def read_data(self):
...
 
-    def _write_register_byte(self, register, value):
+    def _write_register_byte(self, register: int, value: int) -> None:
         """Write a single byte register at the specified register address.
         Subclasses MUST implement this!"""
         raise NotImplementedError
 
     @property
-    def touches(self):
+    def touches(self) -> List[Tuple[int, int, int]]:
```
Comment (tekktrik) [inline adafruit_stmpe610.py]: It's actually appending dictionaries to the list with values from the unpacked tuple, so the return is `List[Dict[str, int]]`.

File: adafruit_stmpe610.py [hunk 3]
```diff
@@ -235,24 +244,24 @@ def get_version(self):
...
     @property
-    def buffer_empty(self):
+    def buffer_empty(self) -> bool:
         """Buffer empty status."""
         empty = self._read_byte(_STMPE_FIFO_STA) & _STMPE_FIFO_STA_EMPTY
         return empty != 0
 
     @property
-    def get_point(self):
+    def get_point(self) -> dict:
```
Comment (tekktrik) [inline adafruit_stmpe610.py]: We can better specify the contents of the dictionary using generics; let's type this as `Dict[str, int]`.

File: adafruit_stmpe610.py [hunk 4]
```diff
@@ -291,13 +300,13 @@ class Adafruit_STMPE610_I2C(Adafruit_STMPE610):
...
-        i2c,
-        address=_STMPE_ADDR,
-        calibration=None,
-        size=None,
-        disp_rotation=0,
-        touch_flip=(False, False),
-    ):
+        i2c: I2C,
+        address: int = _STMPE_ADDR,
+        calibration: Optional[Tuple[int, int]] = None,
```
Comment (tekktrik) [inline adafruit_stmpe610.py]: Looks like `calibration` is a pair of nested tuple:  `Optional[Tuple[Tuple[int, int], Tuple[int, int]]]`.

File: adafruit_stmpe610.py [hunk 5]
```diff
@@ -320,11 +329,13 @@ def __init__(  # pylint: disable=too-many-arguments
...
         if _STMPE_VERSION != version:
-            raise RuntimeError("Failed to find STMPE610! Chip Version 0x%x" % version)
+            raise RuntimeError(
+                f"Failed to find STMPE610! Chip Version {hex(version).upper()}."
+            )
         super().__init__()
 
     @property
-    def touch_point(self):  # pylint: disable=too-many-branches
+    def touch_point(self) -> Tuple[int, int, int]:  # pylint: disable=too-many-branches
```
Comment (tekktrik) [inline adafruit_stmpe610.py]: Since this can also return `None`, it should be `Optional[Tuple[int, int, int]]`.

File: adafruit_stmpe610.py [hunk 6]
```diff
@@ -368,7 +379,7 @@ def touch_point(self):  # pylint: disable=too-many-branches
             return (x, y, pressure)
         return None
 
-    def _read_register(self, register, length):
+    def _read_register(self, register: int, length: int) -> List[int]:
```
Comment (tekktrik) [inline adafruit_stmpe610.py]: While technically `bytearray` **is** a list of `int`, we can be more specific and type it as `bytearray` return.

File: adafruit_stmpe610.py [hunk 7]
```diff
@@ -419,14 +430,14 @@ class Adafruit_STMPE610_SPI(Adafruit_STMPE610):
...
-        baudrate=1000000,
-        calibration=None,
-        size=None,
-        disp_rotation=0,
-        touch_flip=(False, False),
-    ):
+        spi: SPI,
+        cs: Pin,
+        baudrate: int = 1000000,
+        calibration: Optional[Tuple[int, int]] = None,
```
Comment (tekktrik) [inline adafruit_stmpe610.py]: See the other comment about typing `calibration`.

File: adafruit_stmpe610.py [hunk 8]
```diff
@@ -456,14 +467,14 @@ def __init__(  # pylint: disable=too-many-arguments
...
-                    "Failed to find STMPE610 controller! Chip Version 0x%x. "
+                    f"Failed to find STMPE610 controller! Chip Version {hex(version).upper()}. "
                     "If you are using the breakout, verify you are in SPI mode."
                     % version
                 )
         super().__init__()
 
     @property
-    def touch_point(self):  # pylint: disable=too-many-branches
+    def touch_point(self) -> Tuple[int, int, int]:  # pylint: disable=too-many-branches
```
Comment (tekktrik) [inline adafruit_stmpe610.py]: Same thing, this can return `None`.

File: adafruit_stmpe610.py [hunk 9]
```diff
@@ -509,7 +520,7 @@ def touch_point(self):  # pylint: disable=too-many-branches
 
     # pylint: disable=no-member
     # Disable should be reconsidered when refactor can be tested.
-    def _read_register(self, register, length):
+    def _read_register(self, register: int, length: int) -> List[int]:
```
Comment (tekktrik) [inline adafruit_stmpe610.py]: Same thing as before, this returns `bytearray`.

**Review by tekktrik — CHANGES_REQUESTED** [review]

**Comment by tcfranks** [comment]
resubmitted with changes

File: adafruit_stmpe610.py [hunk 10]
```diff
@@ -26,12 +26,19 @@
...
+    from busio import I2C, SPI
+except ImportError:
+    pass
 
 __version__ = "0.0.0+auto.0"
 __repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_STMPE610.git"
 
 
-def map_range(x, in_min, in_max, out_min, out_max):
+def map_range(x: float, in_min: int, in_max: int, out_min: int, out_max: int) -> float:
```
Comment (tekktrik) [inline adafruit_stmpe610.py]: Sorry, didn't realize this the first time, but all of the arguments can conceivably be `float`.

File: adafruit_stmpe610.py (hunk 7 above)
Comment (tekktrik) [inline adafruit_stmpe610.py]: This still needs to be addressed similar to the other `calibration` argument.

**Review by tekktrik — CHANGES_REQUESTED** [review]

File: adafruit_stmpe610.py (hunk 7 above)
Comment (tcfranks) [reply adafruit_stmpe610.py]: yep - missed that one myself

**Comment by tcfranks** [comment]
resubmitted with changes

**Review by tekktrik — APPROVED** [review]
Thanks!

**Pull request merged**